eng/scripts/Sdk-Generate.ps1

```diff
@@ -0,0 +1,173 @@
+param(
+$GenerateInput, 
+$GenerateOutput,
+$RepoRoot = "${PSScriptRoot}/../.."
+)
+
+$input = Get-Content $GenerateInput | ConvertFrom-Json
+$inputFiles = $input.changedFiles;
+$inputFiles += $input.relatedReadmeMdFiles;
+$inputFiles = $inputFiles | select -Unique
+Write-Output "List Of changed swagger files and related readmes" $inputFiles "`n"
+
+$autorestFilesPath = Get-ChildItem -Path "$RepoRoot/sdk"  -Filter autorest.md -Recurse | Resolve-Path -Relative
+
+Write-Host "Updating autorest.md files for all the changed swaggers."
+$sdkPaths = @()
+$headSha = $input.headSha
+$repoHttpsUrl = $input.repoHttpsUrl
+foreach ($inputFile in $inputFiles)
+{
+  foreach ($path in $autorestFilesPath)
+  {
+    $fileContent = Get-Content $path
+    $isUpdatedLines = $false
+    $updatedLines = foreach($line in $fileContent)
+    {
+      if($line -match "[\/][0-9a-f]{4,40}[\/]$inputFile")
+      {
+        # replacing sha
+        $line = $line -replace "[\/][0-9a-f]{4,40}[\/]$inputFile", "/$headSha/$inputFile"
+        # replacing repo path
+        $line -replace "https:\/\/[^`"]+?$headSha", "$repoHttpsUrl/blob/$headSha"
```

Comment:
For regex `+?` -> `*`. Why not do these 2 replacements in the same regex?

Reply:
Updated. I thought it will be more readable if we split it into two.